### ISSUE TITLE
Request for Modal component to accept Top

### ISSUE THREAD

**Issue by sambigelow** [chakra-ui/chakra-ui#631]
**Is your feature request related to a problem? Please describe.**
The modal top is hard coded at the moment, and the only alternative is to pass `isCentered`. It'd be nice to have finer grain control over where the modal shows up on the screen. In our app, this hardcoded number puts the modal directly inline with the header and it looks kind of funny. It'd be nice to be able to change it.

**Describe the solution you'd like**
Be able to pass the `top` prop to the `Modal` component so we can adjust where the modal shows up.

**Describe alternatives you've considered**
Another thing I thought is a prop that is something like `positionOnScreen` that can put it at 0%, 25%, 50%, 75%, or 100%.

I'd be happy to do the work on this. I'm curious about the maintainer opinions before starting.

**Comment by joserocha3** [comment]
I was able to position it by changing the `margin-top` value on the `ModalContent`. Maybe that will meet your needs.

```js
<ChakraModal isOpen={isOpen} preserveScrollBarGap {...rest}>
    <ModalOverlay />
    <ModalContent rounded={5} mt={[0, 50]} bg="white">
      {header}
      <ModalBody>{children}</ModalBody>
      {footer}
    </ModalContent>
  </ChakraModal>
```

**Comment by segunadebayo** [comment]
Thanks for chiming in @joserocha3. @sambigelow, that solution should work!

**Comment by Enigma10** [comment]
But how can i place at bottom ? 
i want my modal to be a bottom when user open website in mobile

**Comment by PierreAndreis** [comment]
> I was able to position it by changing the `margin-top` value on the `ModalContent`. Maybe that will meet your needs.
> 
> ```js
> <Modal isOpen={isOpen} preserveScrollBarGap {...rest}>
>   <ModalOverlay />
>   <ModalContent rounded={5} mt={[0, 50]} bg="white">
>     {header}
>     <ModalBody>{children}</ModalBody>
>     {footer}
>   </ModalContent>
> </Modal>
> ```
This applies margin-top to the content...
![image](https://user-images.githubusercontent.com/5843816/86410931-96c3e100-bc70-11ea-9e83-10c2aa6fbc9b.png)

but the Modal (div parent of ModalContent) still has "top" on it.

![image](https://user-images.githubusercontent.com/5843816/86410970-ac390b00-bc70-11ea-9883-7e07efc94d4d.png)

**Comment by PierreAndreis** [comment]
I actually figured out why the top is still there.
Apparently, if you set the scroll to be inside the modal, it sets a hard-coded top on the modal.

Reproducible: https://codesandbox.io/s/epic-zhukovsky-93lnw?file=/src/App.js

It would be nice if we could remove this top. In mobile, we want the modal to be full screen

**Comment by GraemeFulton** [comment]
using `marginTop="0"` on `ModalContent` worked for me 🤔 with scroll set to inside

**Comment by nomadoda** [comment]
> But how can i place at bottom ? i want my modal to be a bottom when user open website in mobile

Use `alignSelf="flex-end"` on `<ModalContent />`

**Comment by Ilhamhidayatulloh59** [comment]
kmkl

**Comment by ArmaandeepSingh123** [comment]
@segunadebayo 
Please get it replaced to margin-top in the documentation - https://chakra-ui.com/docs/components/modal#make-modal-vertically-centered



![image](https://github.com/chakra-ui/chakra-ui/assets/146326882/81a12a35-3a7c-4f45-a71c-295247893e13)
